cmd/all-in-one/main.go

```diff
@@ -267,6 +268,11 @@ func startQuery(
 	metricsQueryService querysvc.MetricsQueryService,
 	baseFactory metrics.Factory,
 ) *queryApp.Server {
+	if qOpts.Tenancy.Enabled {
+		spanReader = tenancy.NewGuardedSpanReader(spanReader, &qOpts.Tenancy)
+		depReader = tenancy.NewGuardedDependencyReader(depReader, &qOpts.Tenancy)
```

Comment:
nit: tenancy guard is a cross-cutting concern, similar to metrics. We usually do not apply metrics decorators in the main functions, but do it centrally in the storage factory facade. I think it would be better to do similar thing with the tenancy guard decorator, to ensure it's always present, instead of copy-pasting the same across many main()s

Reply:
@yurishkuro The PR for collector had an explicit guard check in PostSpans().  This PR wrapped the spanReader, similar to how Metrics are wrapped in several _main.go_ with `spanReader = spanstoreMetrics.NewReadMetricsDecorator(spanReader, metricsFactory)`.

Do you prefer:
- explicit guard in each method?
- explicit wrapper in each _main.go_ that constructs a spanReader
- Add tenancy to `storage.FactoryConfig`

Reply:
my preference is to enable the guard at the factory level